### PR TITLE
updating Bio.Affy for python3

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request updates Bio.Affy for python3, with lots of cleanup

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
V4 is binary, but I thought the V3 format was text - why insist on parsing it as bytes?

File: Bio/Affy/CelFile.py [hunk 1]
```diff
@@ -94,88 +83,103 @@ def __init__(self):
...
+        raise ValueError("Failed to read magic number from Affy Version 4 CEL file") from None
+    if magicNumber != (64, ):
+        raise ValueError("Incorrect magic number in Affy Version 4 CEL file")
+    return _read_v4(handle)
 
 
 # read Affymetrix files version 4.
 def read_v4(f):
-    """Read verion 4 Affymetrix CEL file, returns corresponding Record object.
+    "Read version 4 Affymetrix CEL file, and return a Record object (DEPRECATED)."
```
Comment (peterjc) [inline Bio/Affy/CelFile.py]: Triple quotes for a docstring.

File: Bio/Affy/CelFile.py [hunk 2]
```diff
@@ -94,88 +83,103 @@ def __init__(self):
...
+    raise Exception("The read_v4 function in Bio.Affy.CelFile is deprecated."
+                    "Instead, please use the read function in Bio.Affy.CelFile "
+                    "specifying version=4.")
 
-    Most importantly record.intensities correspond to intensities from the CEL
-    file.
 
-    record.mask and record.outliers are not set.
+def read_v3(handle):
+    "Read version 3 Affymetrix CEL file, and return a Record object (DEPRECATED)."
```
Comment (peterjc) [inline Bio/Affy/CelFile.py]: Triple quotes for a docstring

**Comment by mdehoon** [comment]
> V4 is binary, but I thought the V3 format was text - why insist on parsing it as bytes?

The `DatHeader` line in the V3 file is a binary value.

**Comment by mdehoon** [comment]
> > V4 is binary, but I thought the V3 format was text - why insist on parsing it as bytes?
> 
> The `DatHeader` line in the V3 file is a binary value.

Actually I take that back. The `DatHeader` line uses `\x14` as a field delimiter but that is still valid ASCII. Reverting to text mode for V3.

**Comment by mdehoon** [comment]
thanks

**Pull request merged**